powershell/plugins/sdk-create-inline-properties.ts

```diff
@@ -367,11 +368,74 @@ async function createVirtuals(state: State): Promise<PwshModel> {
   return state.model;
 }
 
+function addPropertyWithDuplicateNameReverse(properties: Array<Property>, duplicate: Property) {
+  let count = 0;
+  properties.filter(property => {
+    if (property.language.default.name.startsWith(duplicate.language.default.name)) {
+      const name = property.language.default.name.substring(duplicate.language.default.name.length);
+      if (name === '' || typeof Number(name) === 'number') {
+        return true;
+      }
+    }
+    return false;
+  }).sort((a, b) => {
+    const keyA = a.language.default.name.substring(duplicate.language.default.name.length);
+    const keyB = b.language.default.name.substring(duplicate.language.default.name.length);
+    const numA = keyA === '' ? 0 : Number(keyA);
+    const numB = keyB === '' ? 0 : Number(keyB);
+    return numA - numB;
+  }).forEach(property => {
+    const key = property.language.default.name.substring(duplicate.language.default.name.length);
+    const num = key === '' ? 0 : Number(key);
+    if (num === count) {
+      property.language.default.name = duplicate.language.default.name + (++count);
+    }
+  });
+  properties.unshift(duplicate);
+}
+
+function moveAdditionalPropertiesFromParentToProperties(obj: ObjectSchema, schemas: Schemas) {
+  if (obj.parents) {
+    let schema: Schema | undefined;
+    if (Array.isArray(obj.parents.immediate)) {
+      obj.parents.immediate = obj.parents.immediate.filter(parent => {
+        if (parent.type === 'dictionary' && parent.language.default.name === obj.language.default.name && schemas.dictionaries?.find(dictionary => dictionary.language.default.name === parent.language.default.name)) {
+          schema = parent;
+          return false;
+        }
+        return true;
+      });
+    }
+    if (Array.isArray(obj.parents.all)) {
+      obj.parents.all = obj.parents.immediate.filter(parent => {
+        if (parent.type === 'dictionary' && parent.language.default.name === obj.language.default.name && schemas.dictionaries?.find(dictionary => dictionary.language.default.name === parent.language.default.name)) {
+          schema = parent;
```

Comment:
For those that exist in parents.immediate must exist in parents.all. And there are issues as below.

1. It should obj.parents.all instead of obj.parents.immediate in line 410.
2. Line 412 is redundant.
3. Based on my assumption above, can merge line the logic for parents.all with parents.immediate

Reply:
decided to keep redundancy after discussion, because reference of additional properties need to be removed from both all and immediate